### DISCUSSION TITLE
Welcome to WhiteboxTools-ArcGIS Discussions!

### DISCUSSION THREAD

**Discussion by giswqs** [opengeos/WhiteboxTools-ArcGIS#24] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Darell-Arocho** [comment]
Hi! Is it possible to create a landform classification map using this Wb tools? If so, how can I go about it? Like which tools can help me with this?

**Comment by jblindsay** [comment]
Yes it is. You may want to look at the [PennockLandformClassification](https://www.whiteboxgeo.com/manual/wbt_book/available_tools/geomorphometric_analysis.html#PennockLandformClass) tool. There are also several [machine learning based classification tools](https://www.whiteboxgeo.com/whitebox-general-toolset/) in the extensions for Whitebox that are widely used in landform classification applications. By the way, this is a question that likely better belongs on the [WhiteboxTools github repo](https://github.com/jblindsay/whitebox-tools), or better yet, on our [Whitebox Google Groups](https://groups.google.com/g/whiteboxtools?pli=1).

**Reply by Darell-Arocho** [reply]
Thank you so much for the quick reply! I joined the Whitebox Google Groups.

**Comment by SDGIS** [comment]
I have 78K points (representing cobrahead street lights) with angles (representing the angle of the mast arm.) I want to extend a line in the direction of the angle until it hits a line in a different dataset (street centerline.) This would identify which street segment the street light is illuminating. Traffic counts, road width, and roadway classification are all characteristics to determine which light should be installed. Importantly, it isn't the nearest street centerline, but the one the mast arm is facing.

I've looked high and low for this tool, but it doesn't seem to exist. Currently available tools can be used to create this functionality, but they produce unreliable results and/or aren't scaleable.

Bearing Distance To Line requires a distance
Extend Line goes in two directions, and when used in modelbuilder/arcpy doesn't work reliably.

**Comment by jblindsay** [comment]
Take a look at the [ExtendVectorLines](https://www.whiteboxgeo.com/manual/wbt_book/available_tools/gis_analysis.html#ExtendVectorLines) tool. It takes care of the first aspect of what you want, extending the lines in the direction of the angle. It doesn't perform the intersection with other distant lines though. You'd need to work towards a workflow to achieve that.

**Reply by SDGIS** [reply]
Thank you. I solved this by running the Bearing Distance To Line (ExtendVectorLines works the same) at 50 feet, then created a point at the intersection(s) created by the new vector line, split the line at the intersection point, then select from the split lines which intersect my initial points. It isn't perfect. I plan to adjust the tool to run at further distances (maybe 100 feet) on records that didn't reach a centerline in 50 feet. It takes a long time to run. I'm sure you could build something that would run quicker and create more reliable results.

Thank you for WhiteBox tools.
